models/google_ads__url_report.sql

```diff
@@ -0,0 +1,92 @@
+with stats as (
+
+    select *
+    from {{ var('ad_stats') }}
+), 
+
+accounts as (
+
+    select *
+    from {{ var('account_history') }}
+    where is_most_recent_record = True
+), 
+
+campaigns as (
+
+    select *
+    from {{ var('campaign_history') }}
+    where is_most_recent_record = True
+), 
+
+ad_groups as (
+
+    select *
+    from {{ var('ad_group_history') }}
+    where is_most_recent_record = True
+),
+
+ads as (
+
+    select *
+    from {{ var('ad_history') }}
+    where is_most_recent_record = True
+), 
+
+fields as (
+
+    select
+        stats.date_day,
+        accounts.account_name,
+        accounts.account_id,
+        campaigns.campaign_name,
+        campaigns.campaign_id,
+        ad_groups.ad_group_name,
+        ad_groups.ad_group_id,
+        ads.ad_id,
+        ads.base_url,
+        ads.url_host,
+        ads.url_path,
+
+        {% if var('google_auto_tagging_enabled', false) %}
+
+        coalesce( {{ dbt_utils.get_url_parameter('ads.final_url', 'utm_source') }} , 'google')  as utm_source,
+        coalesce( {{ dbt_utils.get_url_parameter('ads.final_url', 'utm_medium') }} , 'cpc') as utm_medium,
+        coalesce( {{ dbt_utils.get_url_parameter('ads.final_url', 'utm_campaign') }} , campaigns.campaign_name) as utm_campaign,
+        coalesce( {{ dbt_utils.get_url_parameter('ads.final_url', 'utm_content') }} , ad_groups.ad_group_name) as utm_content,
+
+        {% else %}
+
+        ads.utm_source,
+        ads.utm_medium,
+        ads.utm_campaign,
+        ads.utm_content,
+        
```

Comment:
Hey Joe, I was just doing some work for Microsoft and was referencing this piece of code since both advertisers utilize the autotagging feature. 

Why are we doing a coalesce here rather than just using `google`, `cpc`, `campaigns.campaign_name`, `ad_groups.ad_group_name`? It looks like we're already using `dbt_utils.get_url_parameter('final_url', 'utm_source')` and etc in the `stg_*` models so this logic seems a little redundant. 

As for `utm_term` it looks like in Microsoft it's defined as the actual keyword terms from the keyword report. Not sure if that's different for Google.